core/layout.go

```diff
@@ -1847,3 +1849,25 @@ func (fr *Frame) scrollResetIfNone() {
 		}
 	}
 }
+
+// DirectRenderDrawBBoxes returns the destination and source bounding boxes
+// for RenderDraw call for widgets that do direct rendering.
+// The destBBox.Min point can be passed as the dp destination point for Draw
+// function, and srcBBox is
+func (wb *WidgetBase) DirectRenderDrawBBoxes(srcFullBBox image.Rectangle) (destBBox, srcBBox image.Rectangle, empty bool) {
```

Comment:
Your documentation trails off here. Also I thought we weren't use `DirectRender` terminology anymore?

Reply:
oops. DirectRender is still used for direct renderers -- but everyone just has a RenderDraw method.